sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -681,16 +705,24 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                         }
                         try
                         {
-                            receiver = await ServiceBusSessionReceiver.CreateSessionReceiverAsync(
-                                entityPath: EntityPath,
-                                connection: _connection,
-                                options: receiverOptions,
-                                cancellationToken: cancellationToken).ConfigureAwait(false);
-
-                            await OnSessionInitializingAsync(
-                                (ServiceBusSessionReceiver)receiver,
-                                cancellationToken).
-                                ConfigureAwait(false);
+                            if (sessionId != null)
+                            {
+                                if (!sessionIdsReceiverMap.ContainsKey(sessionId))
```

Comment:
nit: can we fold this initial sessionIdsReceiverMap check into GetOrCreateSessionReceiver?